redash/query_runner/azure_sql_data_warehouse.py

```diff
@@ -0,0 +1,192 @@
+import json
+import logging
+import sys
+import uuid
+
+from redash.query_runner import *
+from redash.utils import JSONEncoder
+
+logger = logging.getLogger(__name__)
+
+try:
+    import pyodbc
+    enabled = True
+except ImportError:
+    enabled = False
+
+# from _mssql.pyx ## DB-API type definitions & http://www.freetds.org/tds.html#types ##
+types_map = {
+    1: TYPE_STRING,
+    2: TYPE_BOOLEAN,
+    # Type #3 supposed to be an integer, but in some cases decimals are returned
+    # with this type. To be on safe side, marking it as float.
+    3: TYPE_FLOAT,
+    4: TYPE_DATETIME,
+    5: TYPE_FLOAT,
+}
+
+
+class MSSQLJSONEncoder(JSONEncoder):
+    def default(self, o):
+        if isinstance(o, uuid.UUID):
+            return str(o)
+        return super(MSSQLJSONEncoder, self).default(o)
+
+
+class SqlServer(BaseSQLQueryRunner):
+    noop_query = "SELECT 1"
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "user": {
+                    "type": "string"
+                },
+                "password": {
+                    "type": "string"
+                },
+                "server": {
+                    "type": "string",
+                    "default": "127.0.0.1"
+                },
+                "port": {
+                    "type": "number",
+                    "default": 1433
+                },
+                "tds_version": {
+                    "type": "string",
+                    "default": "7.0",
+                    "title": "TDS Version"
+                },
+                "charset": {
+                    "type": "string",
+                    "default": "UTF-8",
+                    "title": "Character Set"
+                },
+                "db": {
+                    "type": "string",
+                    "title": "Database Name"
+                },
+                "driver": {
+                    "type": "string",
+                    "title": "Driver Identifier"
+                }
+            },
+            "required": ["db"],
+            "secret": ["password"]
+        }
+
+    @classmethod
+    def enabled(cls):
+        return enabled
+
+    @classmethod
+    def name(cls):
+        return "Azure SQL Data Warehouse"
```

Comment:
Maybe we should call it "Microsoft SQL Server (ODBC)"?

I saw weird issues with pymssql and Azure (not only Data Warehouse), so I'm thinking of deprecating the other connector and using this one for all SQL Server connections in the future.